README.md

```diff
@@ -55,7 +64,89 @@ DynamicConsumer
   .runDrain
 ```
 
-DynamicConsumer is built on `ZManaged` and therefore resource-safe: after stream completion all resources acquired will be shutdown.
+
+### Notes
+
+- DynamicConsumer is built on `ZManaged` and therefore resource-safe: after stream completion all resources acquired will be shutdown.
+
+- DynamicConsumer.shardedStream takes default value for initialPosition in the stream that the application should 
+  start at = `TRIM_HORIZON`, which is from the oldest messages in Kinesis.
+  However, from the KCL documentation, the initial position is only used during initial lease creation.
+  When an application restarts, it will resume from the previous checkpoint,
+  and so will continue from where it left off in the Kinesis stream.
+
+#### Checkpoint coordination schemes
+
+The handler for messages in the example above calls `r.checkpoint`. This checkpoints every message.
```

Comment:
Let's adjust the example itself and add an inline comment that batched checkpointing is recommended for performance. I think a solution with `aggregateAsync` is the most ZIO-idiomatic solution. If you need help with that I can give it a go.